### PR TITLE
Handle STM32 random failure

### PR DESCRIPTION
**Pull request by AAndrisa**
## Pull Request Description

For the build of STM32, in some cases a random failure appears, to handle this case a new run of the failed project is made.
If the new rerun exists also with error status this will be interpreted as a persistent error not a random one.
In case a project is build twice it is mentioned in the log message.

## PR Type
- [x] Bug fix (change that fixes an issue)
- [ ] New feature (change that adds new functionality)
- [x] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [ ] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [ ] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: tools/scripts/build_projects.py [hunk 1]
```diff
@@ -84,13 +84,17 @@ def shell_source(script):
 
 	os.environ.update(env)
 
+
```
Comment (amiclaus) [inline tools/scripts/build_projects.py:105]: unrelated change.

**Review by amiclaus — COMMENTED** [review]
for commit e36c8c8658a72cf57778a10d4cb17ba8359f002c can you please explain a bit better the changes in the commit body?

**Comment by AAndrisa** [comment]
v2: removed new line from commit, changed commit message

**Review by SRaus — APPROVED** [review]

**Review by amiclaus — APPROVED** [review]
